src/ServiceManagerBase.sol

```diff
@@ -112,14 +107,62 @@ abstract contract ServiceManagerBase is OwnableUpgradeable, ServiceManagerBaseSt
                 strategies[j] = strategy;
             }
 
-            _operatorSetManager.addStrategiesToOperatorSet(
-                uint32(i),
-                strategies  
-            );
+            _operatorSetManager.addStrategiesToOperatorSet(uint32(i), strategies);
+            emit OperatorSetStrategiesMigrated(uint32(i), strategies);
         }
         isStrategiesMigrated = true;
     }
 
+    /**
+     * @notice the operator needs to provide a signature over the operatorSetIds they will be registering
+     * for. This can be called externally by getOperatorSetIds
+     */
+    function migrateOperatorToOperatorSets(
+        address operator,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory signature
+    ) external {
+        require(
+            !isOperatorMigrated[operator],
```

Comment:
https://github.com/Layr-Labs/eigenlayer-contracts/blob/51ff1682890f537915b90abfb5074daedd61949e/src/contracts/core/AVSDirectoryStorage.sol#L51

Reply:
technically we actually don't need this check?

Reply:
Yeah so basically check that operatorAVSOperatorSetCount == 0 but their avsRegistrationStatus is registered. Meaning they haven't migrated yet

Reply:
https://github.com/Layr-Labs/eigenlayer-middleware/pull/263/commits/9880a8de41a12856928e756dd244d1f4b16c75b9#diff-774fbba770093bd7537c705a51e10a3c78f899c50b2aa1bcaa677a99f08acf7eR139-R141